pkg/framework/provider/crypto/crypto.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package crypto provides the default implementation of pkg/framework/spi/crypto.Crypto interface
+//
+// It uses github.com/tink/go crypto primitives
+package crypto
+
+import (
+	"errors"
+
+	"github.com/google/tink/go/aead"
+	"github.com/google/tink/go/core/primitiveset"
+	"github.com/google/tink/go/keyset"
+	"github.com/google/tink/go/signature"
+	aeadsubtle "github.com/google/tink/go/subtle/aead"
+	"golang.org/x/crypto/chacha20poly1305"
+)
+
+// Crypto is the default Crypto implementation
+type Crypto struct {
+}
+
+// Encrypt will encrypt msg using the implementation's corresponding encryption key and primitive in kh
+func (t *Crypto) Encrypt(msg, aad []byte, kh interface{}) ([]byte, []byte, error) {
+	keyHandle, ok := kh.(*keyset.Handle)
+	if !ok {
+		return nil, nil, errors.New("bad key handle format")
+	}
+
+	a, err := aead.New(keyHandle)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ct, err := a.Encrypt(msg, aad)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ps, err := keyHandle.Primitives()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ivSize := nonceSize(ps)
```

Comment:
a comment would be nice on why nonce is built this way.

Reply:
done